gen/installer/aws.py

```diff
@@ -219,13 +218,17 @@ def gen_supporting_template():
         })
 
 
-extra_templates = ['aws/dcos-config.yaml', 'coreos-aws/cloud-config.yaml', 'coreos/cloud-config.yaml']
-
-
 def make_advanced_bunch(variant_args, template_name, cc_params):
+    extra_templates = ['aws/dcos-config.yaml', 'aws/templates/advanced/{}'.format(template_name)]
+    if cc_params['os_type'] == 'coreos':
+        extra_templates += ['coreos-aws/cloud-config.yaml', 'coreos/cloud-config.yaml']
+        cloud_init_implementation = 'coreos'
+    else:
+        cloud_init_implementation = 'canonical'
```

Comment:
can we have an assert here that `cc_params['os_type']` is equal to what we expect here? (I worry someone goes to add another os_type, doesn't realize they should touch this code, things sort of work for mysterious reasons until one day it all explodes with an unrelated change)